metricbeat/scripts/module/metricset/metricset.go.tmpl

```diff
@@ -42,15 +39,14 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-// It returns the event which is then forward to the output. In case of an error, a
-// descriptive error must be returned.
-func (m *MetricSet) Fetch() (common.MapStr, error) {
-
-	event := common.MapStr{
-		"counter": m.counter,
-	}
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(report mb.ReporterV2) {
```

Comment:
As far as I understand, this would not have to be changed even after the deprecation? The old `Fetch()` which is used in most modules stays? As `reporterV2` can do everything that `Fetch` can do (I think) I wonder if we should even get rid of the `Fetch()` interface and only use reporter? It makes the implementation a bit less strict to only return one event at the end, but perhaps having less options to implement things is good?

Reply:
Have less options is a good thing from a code maintainability POV.

I didn't want to go as far as labelling the old `Fetch() MapStr` and `Fetch() []MapStr` methods as deprecated now, but that's an option. It would mean we have *lot's* of code to update.

I think we should use this new `mb.Event` with the ReporterV2 interface for a little bit and see if we like the code flow. If it's good then we can seriously consider deprecating the original Fetch methods.

Reply:
Was not thinking of deprecating it in this PR but more as a discussion we should perhaps have.